### PR TITLE
Adds ability to use inhalers on others

### LINKED ISSUE #18145: Can't use inhalers on anyone but yourself
Inhalers don't have an attack() so there's apparently no way to assist an asthmatic with one.

### PR DESCRIPTION
**Pull request by Pathid**
Fixes #18145

:cl:
* rscadd: Inhalers can be used on other people now, when on help intent.

### CODE REVIEW

File: code/game/objects/items/devices/inhaler.dm [hunk 1]
```diff
@@ -11,10 +11,26 @@
...
+	if(cooldown < world.time - 8)
+		var/mob/living/carbon/human/H = M
+		var/obj/item/mouth_protection = H.get_body_part_coverage(MOUTH)
+		if(mouth_protection)
+			to_chat(user, "<span class='warning'>Remove their \the [mouth_protection] first!</span>")
+			return 1
+		if(!H.hasmouth)
+			to_chat(user, "<span class='warning'>There's nowhere to put \the [src] as [H] lacks a mouth!</span>")
+			return 1
+		playsound(H, 'sound/effects/spray2.ogg', 20, 1)
```
Comment (jknpj) [inline code/game/objects/items/devices/inhaler.dm]: Might as well add a proc for this to avoid copypasta

**Review by jknpj — CHANGES_REQUESTED** [review]

File: code/game/objects/items/devices/inhaler.dm [hunk 2]
```diff
@@ -11,10 +11,26 @@
...
 		cooldown = world.time
+
+/obj/item/device/inhaler/attack(mob/living/M, mob/user)
+	if(!ishuman(M) || user.zone_sel.selecting != "mouth" || user.a_intent != I_HELP)
+		return ..()
+	if(cooldown < world.time - 8)
+		var/mob/living/carbon/human/H = M
+		var/obj/item/mouth_protection = H.get_body_part_coverage(MOUTH)
+		if(mouth_protection)
+			to_chat(user, "<span class='warning'>Remove their \the [mouth_protection] first!</span>")
```
Comment (DamianX) [inline code/game/objects/items/devices/inhaler.dm]: Remove their the gas mask first!

**Comment by DamianX** [comment]
un-copypasta the cooldown as well, maybe give it a define too

File: code/game/objects/items/devices/inhaler.dm [hunk 3]
```diff
@@ -11,10 +11,26 @@
...
 /obj/item/device/inhaler/attack_self(mob/user)
 	if(cooldown < world.time - 8  && ishuman(user))
-		var/mob/living/carbon/human/H = user
-		playsound(user, 'sound/effects/spray2.ogg', 20, 1)
-		H.reagents.add_reagent(ALBUTEROL, 5)
-		cooldown = world.time
+		puff(user)
+
+/obj/item/device/inhaler/attack(mob/living/M, mob/user)
+	if(!ishuman(M) || user.zone_sel.selecting != "mouth" || user.a_intent != I_HELP)
```
Comment (DamianX) [inline code/game/objects/items/devices/inhaler.dm]: Didn't we coerce someone into making defines for zones at some point?

File: code/game/objects/items/devices/inhaler.dm (hunk 2 above)
Comment (Pathid) [reply code/game/objects/items/devices/inhaler.dm]: Okay so it does that with or without the \the. What do I do?

File: code/game/objects/items/devices/inhaler.dm (hunk 3 above)
Comment (DamianX) [reply code/game/objects/items/devices/inhaler.dm]: On second thought, is there anything to gain from forcing the player to select the mouth? What else are you going to do to someone else with an inhaler? Smack them with it?

Comment (MadmanMartian) [reply code/game/objects/items/devices/inhaler.dm]: puff groin

File: code/game/objects/items/devices/inhaler.dm (hunk 2 above)
Comment (MadmanMartian) [reply code/game/objects/items/devices/inhaler.dm]: Try /improper instead?

Comment (DamianX) [reply code/game/objects/items/devices/inhaler.dm]: handled in the discord
`Remove their [mouth_protection.name] first!`

**Review by jknpj — DISMISSED** [review]

File: code/game/objects/items/devices/inhaler.dm [hunk 4]
```diff
@@ -11,10 +11,26 @@
...
+		return ..()
+	if(cooldown < world.time - 8)
+		var/mob/living/carbon/human/H = M
+		var/obj/item/mouth_protection = H.get_body_part_coverage(MOUTH)
+		if(mouth_protection)
+			to_chat(user, "<span class='warning'>Remove their \the [mouth_protection] first!</span>")
+			return 1
```
Comment (ShiftyRail) [inline code/game/objects/items/devices/inhaler.dm]: Why returning `1` (true) if it's a falling case ?

Comment (Pathid) [reply code/game/objects/items/devices/inhaler.dm]: From what I understand, returning 1 means it terminates the click code so it doesn't do anything else. Not entirely sure if it means anything if it's attack and not preattack though. Just imitating how the medical stacks have it to be honest.

Comment (ShiftyRail) [reply code/game/objects/items/devices/inhaler.dm]: That's good enough for me

**Review by DamianX — APPROVED** [review]

**Pull request merged**